### PR TITLE
Add reauthenticate method.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/auth.toit [hunk 1]
```diff
@@ -200,19 +200,62 @@ class Auth:
...
+    to the `auth.users.user_metadata` column. The `user_metadata` should be a JSON
+    object that includes user-specific info, such as their first and last name.
+  - "app_metadata": A custom data object to store the user's application specific
+    metadata.
+    This maps to the `auth.users.app_metadata` column. Only a service role can modify.
+    The `app_metadata` should be a JSON object that includes app-specific info, such as
+    identity providers, roles, and other access control information.
   */
   update_current_user data/Map -> none:
     if client_.session_ == null: throw "No session available."
```
Comment (kasperl) [inline src/auth.toit]: Maybe change this check too?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/auth.toit [hunk 2]
```diff
@@ -200,19 +200,62 @@ class Auth:
...
+  - "password": The user's password.
+  - "phone": The user's phone.
+  - "email_confirm": Confirms the user's email address if set to true. Only a service
+    rule can modify.
+  - "nonce": The nonce sent for reauthentication if the user's password is to be
+    updated. Call `reauthenticate()` to get the nonce first.
+  - "ban_duration": Determines how long a user is banned for. The format for the ban
+    duration follows a strict sequence of decimal numbers with a unit suffix. Valid time
+    units are "ns", "us" (or "µs"), "ms", "s", "m", "h". Setting the ban duration to
+    'none' lifts the ban on the user.
```
Comment (kasperl) [inline src/auth.toit]:
```suggestion
    "none" lifts the ban on the user.
```

Comment (floitsch) [reply src/auth.toit]: done.

File: src/auth.toit (hunk 1 above)
Comment (floitsch) [reply src/auth.toit]: done.

**Pull request merged**